.github/workflows/ci.yml

```diff
@@ -0,0 +1,16 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: cachix/install-nix-action@v27
+      with:
+        github_access_token: ${{ secrets.GITHUB_TOKEN }}
+        enable_kvm: true
+        extra_nix_config: "system-features = nixos-test benchmark big-parallel kvm"
+    - run: nixos-rebuild dry-activate
```

Comment:
Could we do a `build-vm`? We'd get a VM we could test in actions for things like network connectivity.